src/context.ts

```diff
@@ -634,6 +634,47 @@ export default class Context {
 		return this._customFieldKeys();
 	}
 
+	private _customFieldValue(experimentName: string, key: string): JSONValue {
+		const experiment = this._index[experimentName];
+
+		if (experiment != null) {
+			const field = experiment.data.customFieldValues?.find((x) => x.name === key);
+			if (field != null) {
+				switch (field.type) {
+					case "text":
+					case "string":
+						return field.value;
+					case "number":
+						return Number(field.value);
+					case "json":
+						try {
+							return JSON.parse(field.value);
+						} catch (e) {
+							if (field.value === "null") return null;
```

Comment:
I think this test should be done before trying to parse the JSON.

Reply:
Updated :)